app/scripts/modules/core/src/pipeline/config/copyStage/CopyStageModal.tsx

```diff
@@ -0,0 +1,193 @@
+            .map((stage: IStage) => {
+              if (isStrategyConfig(config)) {
+                return { strategy: config.name, stage };
+              } else {
+                return { pipeline: config.name, stage };
+              }
+            });
+        });
+
+        return flatten(nestedStageWrappers);
+      });
+  }
+
+  function renderCopyStageCard(option: Option<string>) {
+    const parsed = JSON.parse(option.value);
+    return <CopyStageCardProps {...parsed} />;
+  }
+
+  function renderCopyStageValue(option: Option<string>) {
+    const parsed = JSON.parse(option.value);
+    return parsed.stage.name;
+  }
+
+  function isStrategyConfig(config: IPipeline | IStrategy): boolean {
+    return 'strategy' in config;
+  }
+
+  function copyStage(command: ICopyStageCommand) {
+    const parsed = JSON.parse(command.selectedStage);
+    closeModal(parsed.stage);
+  }
+
+  return (
+    <>
+      <SpinFormik<ICopyStageCommand>
+        initialValues={initialValues}
+        onSubmit={copyStage}
+        render={formik => (
+          <Modal key="modal" show={true} onHide={() => {}}>
+            <ModalClose dismiss={dismissModal} />
+            <Modal.Header>
+              <h3>Copy Stage</h3>
+            </Modal.Header>
+            <Modal.Body>
+              <div className="copy-stage-modal-body">
+                {error && (
+                  <div className="alert alert-danger">
+                    <p>Could not load stages.</p>
+                  </div>
+                )}
+                <Form name="form" className="form-horizontal">
+                  <FormikFormField
+                    fastField={false}
+                    name="application"
+                    label="From Application"
+                    input={inputProps => (
+                      <ReactSelectInput
+                        {...inputProps}
+                        isLoading={fetchApplications.status === 'PENDING'}
+                        clearable={false}
+                        stringOptions={(fetchApplications.result || []).map(a => a.name)}
+                        onChange={e => setApplication(e.target.value)}
+                        value={application}
+                      />
+                    )}
+                  />
+                  <FormikFormField
+                    fastField={false}
+                    name="selectedStage"
+                    label="Copy Stage"
+                    required={true}
+                    input={inputProps => (
+                      <ReactSelectInput
+                        {...inputProps}
+                        isLoading={fetchStages.status === 'PENDING'}
+                        clearable={false}
+                        disabled={isEmpty(fetchStages.result)}
+                        options={(fetchStages.result || []).map(s => ({
+                          label: s.stage.name,
+                          value: JSON.stringify(s),
```

Comment:
I'd prefer to pass the actual object as the `value` instead of stringifying it.  I see that our typings on ReactSelectInput don't currently allow for anything but a `string`.  I'll see if I can get the generics to work and we can circle back and remove the stringifying in a different PR.